connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -436,15 +436,15 @@ public void tick() {
         //       Another example: if multiple configurations are submitted for the same connector,
         //       the only one that actually has to be written to the config topic is the
         //       most-recently one.
-        long nextRequestTimeoutMs = Long.MAX_VALUE;
+        long scheduledTick = Long.MAX_VALUE;
         while (true) {
             final DistributedHerderRequest next = peekWithoutException();
             if (next == null) {
                 break;
             } else if (now >= next.at) {
                 requests.pollFirst();
             } else {
-                nextRequestTimeoutMs = next.at - now;
+                scheduledTick = Math.min(scheduledTick, next.at);
```

Comment:
We will only reach this line if `scheduledTick` is `Long.MAX_VALUE`:

```suggestion
                scheduledTick = next.at;
```